test/robustness/within/quadtree_partition.hpp

```diff
@@ -0,0 +1,149 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2023 Barend Gehrels, Amsterdam, the Netherlands.
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_EXPERIMENTAL_QUADTREE_PARTITION_HPP
+#define BOOST_GEOMETRY_EXPERIMENTAL_QUADTREE_PARTITION_HPP
+
+#include <boost/range.hpp>
+#include <boost/geometry/algorithms/detail/partition.hpp>
+
+#include <functional>
+
+namespace boost { namespace geometry
+{
+
+namespace experimental
+{
+
+///! Adapts a lambda of the main visitor to the two-ranges partition algorithm
+template <typename F>
+struct adapt_partition_visitor
+{
+    const F& m_f;
+
+    explicit adapt_partition_visitor(const F& f)
+        : m_f(f)
+    {}
+
+    template<typename T1, typename T2>
+    bool apply(T1 const& i1, T2 const& i2) const
+    {
+        return m_f(i1, i2);
+    }
+};
```

Comment:
All of the adaptors could probably be replaced with one simply forwarding parameters to function objects:
```
template <typename F>
struct adapt_partition_visitor
{
    F m_f;

    explicit adapt_partition_visitor(F&& f)
        : m_f(std::move(f))
    {}

    template <typename ...Ts>
    auto apply(Ts&& ...is) const
    {
        return m_f(std::forward<Ts>(is)...);
    }
};
```

Reply:
Indeed, works perfectly. Thanks!
:heavy_check_mark: